tests/js/t16822.nim

```diff
@@ -0,0 +1,29 @@
+block: # bug #16822
```

Comment:
how about:
* move these to `tests/tuples/ttuples_issues.nim` instead of creating a new js-only test
* change spec to:
```nim
discard """
  targets: "c cpp js"
"""
```
(also cpp here because tuple have often caused bugs specific to cpp in past)
* remove the echo in this test by changing last few lines to:
block: # bug #14911
  doAssert $(a: 1) == "(a: 1)" # works
  doAssert $(`a`: 1) == "(a: 1)" # works
  doAssert $(`a`: 1, `b`: 2) == "(a: 1, b: 2)" # was: Error: named expression expected
```

(but if you prefer we can also do this in a followup PR)

more generally, IMO we should reserve `tests/js` to things that are really only relevant to js, so we increase test coverage on all backends

Reply:
=> https://github.com/nim-lang/Nim/pull/16907